File: backend/models/plant.py
from datetime import datetime

from sqlmodel import Field, SQLModel


class Plant(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str
    sector: str
    sub_sector: str
    commissioning_date: datetime
    powerMW: float
    min_powerMW: float
    gps: str
    city: str
    departement: int
    region: str

File: frontend/src/pages/plant.ts
export interface Plant {
  name: string
  sector: string
  subSector: string
  gps: string
  city: string
  departement: number
  region: string
  reactors: Reactor[]
}

export interface Reactor {
  id: number
  name: string
  sector: string
  subSector: string
  gps: string
  city: string
  departement: number
  region: string
  reactors: Reactor[]
  commissioningDate: string
  powerMW: number
}
